tests/SAML2/AuthnRequestTest.php

```diff
@@ -345,4 +345,167 @@ public function testRequesterIdIsReadCorrectly()
+
+        $nip = array("Format" => "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+            "SPNameQualifier" => "https://sp.example.com/SAML2",
+            "AllowCreate" => true);
+        $request->setNameIDPolicy($nip);
+
+        $expectedStructureDocument = <<<AUTHNREQUEST
+<samlp:AuthnRequest
+    xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"
+    xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion"
+    ID="123"
+    Version="2.0"
+    IssueInstant="2004-12-05T09:21:59Z"
+    Destination="https://tiqr.example.org/idp/profile/saml2/Redirect/SSO">
+    <saml:Issuer>https://gateway.example.org/saml20/sp/metadata</saml:Issuer>
+    <samlp:NameIDPolicy
+        Format="urn:oasis:names:tc:SAML:2.0:nameid-format:transient"
+        SPNameQualifier="https://sp.example.com/SAML2" AllowCreate="true"
+    />
+</samlp:AuthnRequest>
+AUTHNREQUEST;
+
+        $expectedStructure = DOMDocumentFactory::fromString($expectedStructureDocument)->documentElement;
+        $requestStructure = $request->toUnsignedXML();
+
+        $this->assertEqualXMLStructure($expectedStructure, $requestStructure);
+
+        $this->assertXmlStringEqualsXmlString($expectedStructure->ownerDocument->saveXML(), $requestStructure->ownerDocument->saveXML());
+    }
+
+    /**
+     * Test setting NameIDPolicy with only a Format results in expected XML
+     */
+    public function testSettingNameIDPolicyFormatOnly()
+    {
+        // basic AuthnRequest
+        $request = new AuthnRequest();
+        $request->setIssuer('https://gateway.example.org/saml20/sp/metadata');
+        $request->setDestination('https://tiqr.example.org/idp/profile/saml2/Redirect/SSO');
+        $request->setIssueInstant( Utils::xsDateTimeToTimestamp('2004-12-05T09:21:59Z'));
+
+        $nip = array("Format" => "urn:oasis:names:tc:SAML:2.0:nameid-format:transient");
+        $request->setNameIDPolicy($nip);
+
+        $expectedStructureDocument = <<<AUTHNREQUEST
+<samlp:AuthnRequest
+    xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"
+    xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion"
+    ID="123"
+    Version="2.0"
+    IssueInstant="2004-12-05T09:21:59Z"
+    Destination="https://tiqr.example.org/idp/profile/saml2/Redirect/SSO">
+    <saml:Issuer>https://gateway.example.org/saml20/sp/metadata</saml:Issuer>
+    <samlp:NameIDPolicy Format="urn:oasis:names:tc:SAML:2.0:nameid-format:transient"/>
+</samlp:AuthnRequest>
+AUTHNREQUEST;
+
+        $expectedStructure = DOMDocumentFactory::fromString($expectedStructureDocument)->documentElement;
+        $requestStructure = $request->toUnsignedXML();
+
+        $this->assertEqualXMLStructure($expectedStructure, $requestStructure);
+
+        $this->assertXmlStringEqualsXmlString($expectedStructure->ownerDocument->saveXML(), $requestStructure->ownerDocument->saveXML());
+    }
+
+    /**
+     * Test setting NameIDPolicy with invalid type for AllowCreate.
+     */
+    public function testSettingNameIDPolicyToIncorrectTypeAllowCreate()
```

Comment:
You are only checking for one illegal type. What happens if I throw in an integer, object, resource, array?

We often use a [dataprovider method](https://github.com/SURFnet/Stepup-Middleware/blob/fe9716c6409a7dddaee1753160744da611e6705e/src/Surfnet/Stepup/Tests/Identity/Value/TimeFrameTest.php#L55-L69) (or [class](https://github.com/OpenConext/OpenConext-profile/blob/develop/src/OpenConext/Profile/Tests/DataProvider.php)) for this.

Not saying that it should be changed, but it would be nice to tackle possible cases of (implicit) type conversion this way.